pkg/deploy/generator/resources.go

```diff
@@ -1089,6 +1052,124 @@ EOF
 
 chcon -R system_u:object_r:var_log_t:s0 /var/opt/microsoft/linuxmonagent
 
+cat >/etc/systemd/system/download-mdm-credentials.service <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/download-credentials.sh mdm
+EOF
+
+cat >/etc/systemd/system/download-mdm-credentials.timer <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Timer]
+OnBootSec=0min
+OnCalendar=0/12:00:00
+AccuracySec=5s
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/local/bin/download-credentials.sh <<EOF
+#!/bin/bash
+set -eu
+
+COMPONENT="\$1"
+if [[ "\$COMPONENT" = "mdm" || "\$COMPONENT" = "mdsd" ]]; then
+  NEW_CERT_FILE="/etc/temp-\${COMPONENT}.pem"
+  CURRENT_CERT_FILE="/etc/\${COMPONENT}.pem"
+  SECRET_NAME="rp-\${COMPONENT}"
+  AZURE_CONFIG_DIR="/home/cloud-user/.azure-\${COMPONENT}"
```

Comment:
Please don't use /home/cloud-user -- this script doesn't run as cloud-user.
Ideally I would prefer an approach where temporary files weren't written to /etc.

Personally I'd use `mktemp -d` to create a temporary directory both for use via AZURE_CONFIG_DIR and for the temporary download + `trap EXIT` to ensure that it is removed when done.

Reply:
and the operator who ssh to the vm instance will run with cloud-user?